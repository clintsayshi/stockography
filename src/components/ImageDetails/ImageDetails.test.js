import React from "react";
import renderer from "react-test-renderer";
import { render, screen, cleanup } from "@testing-library/react";
import ImageDetails from "./ImageDetails";

const self = {
  id: 132132,
  tags: "nature, sunrise",
  largeImageURL:
    "https://cdn.pixabay.com/photo/2022/04/21/20/49/road-7148369_960_720.jpg",
  downloads: 22233,
  likes: 32,
  views: 23222,
  imageSize: 4344,
  imageHeight: 3233,
  imageWidth: 3423,
  pageURL: "",
  user: "Mike Jack",
  userImageURL: "",
};

afterEach(() => {
  cleanup();
});

it("renders Image Details successfully", () => {
  render(<ImageDetails self={self} />);
  const component = renderer.create(<ImageDetails self={self} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const imageDetailsEl = screen.getByTestId("image-details");
  expect(imageDetailsEl).toBeInTheDocument();
});
