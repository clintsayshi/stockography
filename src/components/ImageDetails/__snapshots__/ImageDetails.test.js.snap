// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Image Details successfully 1`] = `
<div
  data-testid="image-details"
>
  <section
    className="container mx-auto px-4 md:px-8 lg:px-12 xl:px-16"
  >
    <img
      alt="nature, sunrise"
      src="https://cdn.pixabay.com/photo/2022/04/21/20/49/road-7148369_960_720.jpg"
    />
    <div
      className="py-4"
    >
      <div
        className="flex my-2 gap-4 items-center"
      >
        <img
          alt="Mike Jack"
          className="rounded-full w-10 h-10"
          src=""
        />
        <h1
          className="text-xl font-medium"
        >
          Mike Jack
        </h1>
      </div>
      <ul
        className="flex gap-2 items-center flex-wrap"
      >
        <li
          className="flex gap-2"
        >
          <span
            className="p-1 rounded-md bg-gray-100"
          >
            nature
          </span>
          <span
            className="p-1 rounded-md bg-gray-100"
          >
             sunrise
          </span>
        </li>
      </ul>
      <ul
        className="flex flex-wrap  items-center gap-4 py-2"
      >
        <li
          className="flex items-center"
          title="number of views"
        >
          23222
           
          <span>
            views
          </span>
        </li>
        <li
          className="flex gap-2 items-center"
          title="number of likes"
        >
          <span>
            32
          </span>
          <svg
            className="h-5 w-5"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 10.5a1.5 1.5 0 113 0v6a1.5 1.5 0 01-3 0v-6zM6 10.333v5.43a2 2 0 001.106 1.79l.05.025A4 4 0 008.943 18h5.416a2 2 0 001.962-1.608l1.2-6A2 2 0 0015.56 8H12V4a2 2 0 00-2-2 1 1 0 00-1 1v.667a4 4 0 01-.8 2.4L6.8 7.933a4 4 0 00-.8 2.4z"
            />
          </svg>
        </li>
        <li
          className="flex gap-2 items-center"
          title="number of downloads"
        >
          <span>
            22233
          </span>
          <svg
            className="h-5 w-5"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M3 17a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zm3.293-7.707a1 1 0 011.414 0L9 10.586V3a1 1 0 112 0v7.586l1.293-1.293a1 1 0 111.414 1.414l-3 3a1 1 0 01-1.414 0l-3-3a1 1 0 010-1.414z"
              fillRule="evenodd"
            />
          </svg>
        </li>
        <li>
          <a
            className="underline"
            href=""
            rel="noreferrer"
            target="_blank"
          >
            View on pixabay
          </a>
        </li>
      </ul>
    </div>
  </section>
  <div
    className="container mx-auto mb-12 my-2 px-4 md:px-8 lg:px-12 xl:px-16"
  >
    <p
      className="font-mono"
    >
      width: 
      3423
    </p>
    <p
      className="font-mono"
    >
      Height: 
      3233
    </p>
    <p
      className="font-mono"
    >
      Size: 
      4344
    </p>
  </div>
</div>
`;
